Add a draft for annex encryption usecase

mslw opened the pull request:
This use demonstrates git-annex encryption in a workflow based on RIA stores and temporary clones.

This is based on a workflow we created for data incoming through a web form. A lot of credit for brainstorming the solution goes to @christian-monch, @bpoldrack, @aqw and others. I simplified a little, so that the workflow has two locations and can be easily divided into three parts.

The text covers the entire use, but can be seen as a rough draft in that it needs proofreading (both text style and DataLad correctness), formatting adjustment, and most likely extension/clarification of details. I won't be able to work on it in the next 2 weeks, so if anyone wants, please feel free to commit directly.

adswa commented:
This is a really cool write-up, thanks a lot @mslw for tackling this. I have pushed a few commits on top of yours already, and would be happy to see others giving it  a read as well.

mih commented on docs/usecases/encrypted_annex.rst:
```diff
@@ -0,0 +1,300 @@
+Since we cloned the superdataset from local RIA store, also the subdataset has the `origin` (:term:`Git` :term:`remote`) pointing to that store.
+It also has the ``local-storage`` and ``entrystore-storage`` :term:`sibling`\s; these are the
+git-annex :term:`special remote`\s for the local and remote RIA stores, respectively.
+But to learn about new files that were added in the remote server since we last cloned from there, we need the Git remote.
```
This will become unecessary https://github.com/datalad/datalad-next/issues/170

mih commented on docs/usecases/encrypted_annex.rst:
```diff
@@ -0,0 +1,300 @@
+
+   $ datalad update --sibling entrystore --how merge
+
+.. find-out-more:: A note to users of python API
```
```suggestion
.. find-out-more:: A note to users of Python API
```

mih reviewed (commented):
This is great! Thx!

I left a bunch of comments, including style. In general, it would be nice to give capitalization of names another pass (pointed some out), and change the language to treat "data" as plural.

I link some issues to point out how this workflow would be simplified in the future. It might be worth including the links in the text, such that readers can find them until then.

mih commented:
One more: You could consider using `runrecord` directives to make sure that the code in this write-up continues to run.

adswa commented:
I was about to release the handbook, but realized that it would be best if this usecase would be a part of that release. Thus I will probably postpone a release into the next year.  In the new year, there are more eyes to review this, it is not pressing time wise to have a handbook release, and another advantage is that this would also give metalad the chance to get documented. This isn't meant as any time pressure, just an FYI :)

adswa reviewed (approved):
Thanks for the updates! I agree that the sample paths instead of full run records are sufficient to understand the usecase. During reading, I had the slight feeling that a sketch of the setup (of the involved places in the workflow) could be included for an easier overview of everything. In case you have one lying around from the SFB work you're doing, it would be cool to include it here. But if not, I think it also works fine without one - so it could be good to go from my side. I'll wait a bit with a merge to give @mih a chance for a re-review as well, though :)

mslw commented on docs/usecases/encrypted_annex.rst:
```diff
@@ -0,0 +1,319 @@
+     --new-store-ok --name localstore \
+     --alias derived \
+     ria+file:///data/project/store
+   $ git annex enableremote \
+     localstore-storage \
+     keyid+=2EF857223E033AF1
```
I think it's missing encryption type declaration here & below - may be that hybrid is the default, but even if that's the case it's probably better to be explicit
```suggestion
   $ git annex enableremote \
     localstore-storage \
     encryption='hybrid' \
     keyid+=2EF857223E033AF1
```

mslw replied:
Yes, I must have missed it. Fixed in 0668a40

mslw commented:
I agree with you @adswa that a better explanation of the setup was needed. I added two paragraphs at the beginning which should briefly and more clearly explain why we use "remote server" and "local machine" and why we set them up the way we do. I also fixed two `enableremote` commands where I forgot to add encryption mode (necessary).

adswa commented:
Thanks for the description, @mslw! I finally found the time to add an illustration to it as well. Would you mind checking if it is roughly how you would envision a sketch of the workflow?
I'm also giving @mih another ping here.

mslw commented:
The illustration looks good to me, thank you for preparing it!

For the record, this is what I had in mind when working on the thing:

![PXL_20230118_172829676~2](https://user-images.githubusercontent.com/11985212/213292216-144ac69d-3bbe-498d-b44f-02b0cfa711a7.jpg)

But I think for the handbook your version works better. If I were to be picky, I would maybe swap places of the dataset and RIA within the cloud to highlight that the clone owned by the person on the right is made from RIA, and not the dataset checkout (i.e. form -> dataset -> RIA -> another dataset -> another RIA). And maybe make the small arrow between dataset and RIA on the right computer go both ways. But this is being picky, and the things can also stay as they are.

adswa commented:
As a heads-up: I'll merge this later today
